Source/Tools/LinuxEmulation/LinuxSyscalls/ThreadManager.h

```diff
@@ -22,6 +22,82 @@ namespace FEX::HLE {
 class SyscallHandler;
 class SignalDelegator;
 
+// A latch that can be inspected to see if there is a waiter currently active.
+// In contrast to std::condition_variable, an InspectableLatch may be signaled *before* being waited on without change of effect.
+class InspectableLatch final {
+public:
+  bool Wait(struct timespec* Timeout = nullptr) {
+    LOGMAN_THROW_A_FMT(Mutex.load() != HAS_SIGNALED, "Stale signal mutex!");
```

Comment:
This line breaks pre-signaled latches, doesn't it?

Reply:
It specifically doesn't support pre-signaled latches. That's one of the two reasons why I implemented this.

Reply:
Why did you copy my docstring proposal if pre-signaling is not supported? The two are in direct contradiction as-is.

> That's one of the two reasons why I implemented this.

You implemented InspectableLatch in order to be able to... not support something? Why?

(FTR I'd suggest not live-iterating the docstring any further until we find a suitable alternative, since I feel like I'm understanding increasingly less what this class is trying to achieve but GitHub makes going back to your older revisions a pain.)